visualize-admin/visualization-tool
# fix: Minimum dashboard chart sizes

Fixes #1732

This PR:
- fixes some issues with tab labels in layouting step,
- fixes computations of minimal height of free canvas when adding a new chart to an existing layout,
- increases minimal height of free canvas chart from 100 to 150 pixels, which should help with wrong initializations in some cases,
- improves Table appearance in free canvas layout (correct initial sizing and non-overflowing content).

There's still a potential to improve the table chart in free canvas further, as it looks that compact view doesn't work as expected in small sizes and scrollbar position doesn't seem right, but this can be done in another PR, as it seems like a table issue, and not free canvas-related problem.

## How to test
1. Go to [this link](https://visualization-tool-git-fix-minimum-dashboard-chart-sizes-ixt1.vercel.app/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/10&dataSource=Prod).
2. Add a new table chart.
3. ✅ See that the active tab's chart icon is blue, and inactive icon is black.
4. Proceed to layout options.
5. ✅ See that the Label tab is not available under the Title & Description section in the sidebar.
6. ✅ See that the chart icons are visible in the tabs.
7. Switch to a dashboard / free canvas layout.
8. ✅ See that the charts were resized correctly.
9. Go back to the editing mode.
10. Add a new, pie chart.
11. Proceed to layout options.
12. ✅ See that the pie chart was correctly initialized.

## How to reproduce
1. Go to [this link](https://test.visualize.admin.ch/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/10&dataSource=Prod&flag__layouter.dashboard.free-canvas=true).
2. Add a new table chart.
3. ❌ See that the tabs are both gray.
4. Proceed to layout options.
5. ❌ See that the Label tab is available under the Title & Description section in the sidebar.
6. ❌  See that the chart icons are missing from the tabs.
7. Switch to a dashboard / free canvas layout.
8. See that the charts were not resized correctly (e.g. table is not visible at all).
9. Go back to the editing mode.
10. Add a new, pie chart.
11. Proceed to layout options.
12. ❌ See that the pie chart was not correctly initialized.
